lib/pbench/agent/tool_data_sink.py

```diff
@@ -1162,9 +1162,9 @@ def record_tms(self, tms):
         mdlog.set(section, "hostname", self.hostname)
         mdlog.set(section, "hostname-s", hostdata["s"])
         mdlog.set(section, "hostname-f", hostdata["f"])
-        mdlog.set(section, "hostname-i", hostdata["i"])
+        mdlog.set(section, "hostname-i", hostdata["i"].replace("%", "%%"))
```

Comment:
Why might some versions of hostname have `%` but not others?

Are there actually any metadata values we'd log where we _would_ want `ConfigParser` substitutions? Maybe this calls for a `ConfigParser` subclass (e.g., `MetadataParser`) that overrides `set` to quote the strings consistently against future substitution?

Reply:
We might be able to use `interpolation=False` (see https://stackoverflow.com/questions/2537929/is-it-possible-to-temporarily-disable-pythons-string-interpolation).

Reply:
Yes:  it looks like `ConfigParser` provides several options for avoiding interpolation (at the individual `get()` level, at the instance initialization level, and there's also a separate class definition).